lib/thrift/protocol/compact/container_item_type.ex

```diff
@@ -0,0 +1,30 @@
+defmodule Thrift.Protocol.Compact.ContainerItemType do
+  @moduledoc """
+  Type encoding values for maps, lists, and sets differ in compact protocol to those used
+  in struct fields. See https://github.com/apache/thrift/blob/master/doc/specs/thrift-compact-protocol.md
+
+  Note that contrary to the above link, the type code for Bool seems to be 1, not 2,
+  in experiments with the Thrift Ruby implementation.
+  """
+
+  @typedoc "Compact protocol container item type identifiers"
+  @type t :: 1 | 3 | 4 | 6 | 8 | 10 | 11 | 12 | 13 | 14 | 15
+
+  defmacro bool,      do: 1
```

Comment:
According to https://github.com/apache/thrift/blob/master/doc/specs/thrift-compact-protocol.md#list-and-set bools are encoded as 2 within a map, list, or set. But the Ruby thrift implementation encodes them as 1.